fiftyone/utils/huggingface.py

```diff
@@ -1024,12 +1083,27 @@ def _load_fiftyone_dataset_from_config(config, **kwargs):
     splits = _parse_split_kwargs(**kwargs)
 
     download_dir = _get_download_dir(config._repo_id, **kwargs)
-    hfh.snapshot_download(
-        repo_id=config._repo_id, repo_type="dataset", local_dir=download_dir
-    )
+
+    init_download_kwargs = {
+        "repo_id": config._repo_id,
+        "repo_type": "dataset",
+        "local_dir": download_dir,
+    }
 
     dataset_type_name = config._format.strip()
 
+    if dataset_type_name == "FiftyOneDataset":
```

Comment:
Would it make sense to download the images separately only if `max_samples` was provided?

I'm wondering if this might be less efficient:

```py
hfh.snapshot_download(**init_download_kwargs, allow_patterns=[very, long, list, of patterns])
```

than this:

```py
hfh.snapshot_download(**init_download_kwargs)
```